src/main/webapp/app/tasks/task-form/task-form.component.ts

```diff
@@ -14,17 +14,25 @@ import { TaskService } from '../task.service';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class TaskFormComponent {
+  startDate = new Date();
 
   @Output() created: EventEmitter<Task> = new EventEmitter();
 
   taskForm: FormGroup = new FormGroup({
-    name: new FormControl('', Validators.required)
+    name: new FormControl('', Validators.required),
+    dueDate: new FormControl(''),
+    time: new FormControl('')
   });
 
   constructor(@Inject('TaskService') private taskService: TaskService) { }
 
   onSubmit(): void {
-    this.taskService.create(this.taskForm.value.name).subscribe(task => {
+    if(this.taskForm.value.dueDate && this.taskForm.value.time){
```

Comment:
I would use the patchValue method here and move the setting of the time to date into a specific private method.  Like:
`  onSubmit(): void {
    if (this.taskForm.value.dueDate && this.taskForm.value.time) {
      this.taskForm.patchValue({dueDate: this.setTimeToDate(this.taskForm.value.dueDate, this.taskForm.value.time)});
    }
    this.taskService.create(this.taskForm.value.name, this.taskForm.value.dueDate).subscribe(task => {
      this.created.emit(task);
      this.taskForm.reset();
    });
  }

  private setTimeToDate(date: Date, time: string): string {
    const hour = +time.split(':')[0];
    const minute = +time.split(':')[1];
    date.setHours(hour, minute);
    return date.toISOString();
  }`